### PR TITLE
Add Woodwork Craft Shop

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
Content "Woodwork craft shop"

#### Purpose of change
Adds a woodwork variety of craft shop.

#### Describe the solution
Ground floor features a potential pickup truck spawn with nested crates out back, a display window up front with a potential canoe spawn, and a workshop room with tools. The first floor has a computer room with light machinery to be CAM/3D printer for prototyping as well as a rest area, cleaning/eating area and a toilet.

#### Describe alternatives you've considered
The main mapgen file has odd naming but is inline with the other craft shops. 

#### Testing
Ran game with dramatically increased spawnrate of the building and debug teleported around to various instances, everything seemed good.

#### Additional context
First PR so apologies if I've done anything dumb, feedback greatly welcomed.

### CODE REVIEW

**Review by curstwist — COMMENTED** [review]
Can you post some screenshots as well.  Your symbol choices are a little difficult to read in the json.
I don't think you need the `t_null`s in the nested chunks, it should just pick up the floor from the mapgen file.  

Some of the spawn rates seem high, like the trashcans.

File: data/json/mapgen/Woodworker.json [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        "R": "f_locker",
+        "s": "f_sink",
+        "t": "f_table"
+      },
+      "toilets": { "p": {  } },
+      "items": {
+        "b": { "item": "trash", "chance": 70, "repeat": [ 2, 5 ] },
+        "F": { "item": "fridge", "chance": 70, "repeat": [ 2, 10 ] },
+        "k": [ { "item": "file_room", "chance": 40, "repeat": [ 1, 5 ] }, { "item": "writing_utensils", "chance": 30 } ],
+        "s": [ { "item": "SUS_kitchen_sink", "chance": 20, "repeat": [ 2, 6 ] }, { "item": "cleaning", "chance": 20, "repeat": [ 2, 6 ] } ],
```
[the remark this answers is not in the document]
Comment (Procyonae) [reply data/json/mapgen/Woodworker.json]: I changed the itemgroup I were using and totally forgot to remove the repeat, thanks

**Comment by Procyonae** [comment]
> Can you post some screenshots as well. Your symbol choices are a little difficult to read in the json.
> I don't think you need the `t_null`s in the nested chunks, it should just pick up the floor from the mapgen file.
> 
> Some of the spawn rates seem high, like the trashcans.

It won't allow me to either use an empty terrain object or to omit terrain entirely while using rows, I could use place_items and place_furniture instead and remove the rows but I think that would be less clear. I changed the target of t_null to " " to make it clearer that it's just to make it work.

<details>
  <summary>Screenshots</summary>
  
image
image
image
  
</details>

File: data/json/itemgroups/art_antiques_crafts.json [hunk 2]
```diff
@@ -274,5 +274,19 @@
...
+	"id": "woodcrafts",
+	"items": [
+	  [ "holy_symbol_wood", 20],
+	  [ "hand_paddles", 10],
+	  [ "pipe_tobacco", 10],
+	  [ "ukulele", 10],
+	  [ "clogs", 20],
+	  [ "canteen_wood", 10],
+	  [ "pet_carrier_wooden", 10],
+	  [ "wooden_barrel", 10]
```
[the remark this answers is not in the document]
Comment (Procyonae) [reply data/json/itemgroups/art_antiques_crafts.json]: Done, cheers

**Comment by curstwist** [comment]
> 
> It won't allow me to either use an empty terrain object or to omit terrain entirely while using rows, I could use place_items and place_furniture instead and remove the rows but I think that would be less clear. I changed the target of t_null to " " to make it clearer that it's just to make it work.

Odd, I'll try to take a look at it tonight.  Glad it works now.  I like the design with the upper floor being an L.

Does that mechanism work to open the gate?  Its the one for concrete adjacent walls, I'm not sure it'll work with the green walls.

**Comment by Procyonae** [comment]
> > It won't allow me to either use an empty terrain object or to omit terrain entirely while using rows, I could use place_items and place_furniture instead and remove the rows but I think that would be less clear. I changed the target of t_null to " " to make it clearer that it's just to make it work.
> 
> Odd, I'll try to take a look at it tonight. Glad it works now. I like the design with the upper floor being an L.
> 
> Does that mechanism work to open the gate? Its the one for concrete adjacent walls, I'm not sure it'll work with the green walls.

Yeah the gate works fine, didn't realise the mechanisms were wall specific past aesthetics but I guess green walls must inherit from concrete.

**Comment by jbytheway** [comment]
> It won't allow me to either use an empty terrain object or to omit terrain entirely while using rows, I could use place_items and place_furniture instead and remove the rows but I think that would be less clear. I changed the target of t_null to " " to make it clearer that it's just to make it work.

Now that you've changed it to `" "` you should be able to get rid of the `t_null`.  There's special handling for `" "` and `"."` where they are the only two symbols you're allowed to not define any meanings for.

**Pull request merged**